src/allmydata/scripts/create_node.py

```diff
@@ -80,7 +83,7 @@ def validate_where_options(o):
     else:
         # no --location and --port? expect --listen= (maybe the default), and
         # --listen=tcp requires --hostname. But --listen=none is special.
-        if o['listen'] != "none":
+        if o['listen'] != "none" and o.get('join', None) is None:
```

Comment:
I think this implies that "a dict with some keys, you know what they are, right?" as a data-type is failing `validate_where_options`.  It would be pretty cool to introduce something a little bit better defined here.

Not necessarily part of (or blocking) this PR, since the problem is clearly pre-existing...